src/llmo-customer-analysis/utils.js

```diff
@@ -67,21 +67,40 @@ export function getLastSunday() {
   return lastSunday;
 }
 
-function deepEqual(a, b) {
+/**
+ * Compares two prompt arrays for equality, treating them as sets (order-independent)
+ */
+function arePromptArraysEqual(prompts1, prompts2) {
+  if (!Array.isArray(prompts1) || !Array.isArray(prompts2)) return false;
+  if (prompts1.length !== prompts2.length) return false;
+
+  const sorted1 = JSON.stringify(
+    prompts1.sort((a, b) => a.prompt.localeCompare(b.prompt)),
+  );
+  const sorted2 = JSON.stringify(
+    prompts2.sort((a, b) => a.prompt.localeCompare(b.prompt)),
+  );
+
+  return new Set(sorted1).isSupersetOf(new Set(sorted2));
```

Comment:
nit: it is even simpler ;-) 

```suggestion

  return new Set(prompts1).isSupersetOf(new Set(prompts2));
```